src/components/TextInput.tsx

```diff
@@ -0,0 +1,177 @@
+import PropTypes, { ValidationMap } from "prop-types";
+import React from "react";
+import { Hint, HintProps } from "./Hint";
+import { ErrorMessage, ErrorMessageProps } from "./ErrorMessage";
+import {
+  FormGroup,
+  FormGroupPropsWithoutChildren,
+  formGroupWithoutChildrenPropTypes
+} from "./FormGroup";
+import { Label, LabelProps } from "./Label";
+import { Attributes, DataAttributes } from "../helpers/Attributes";
+import classNames from "classnames";
+import "lbh-frontend/lbh/components/lbh-input/_input.scss";
+
+export enum TextInputType {
+  Text = "text",
+  Email = "email",
+  Password = "password",
+  Number = "number",
+  Search = "search",
+  Tel = "tel",
+  Url = "url"
+}
+
+export interface TextInputProps extends React.AriaAttributes, DataAttributes {
+  /**
+   * Id of the input.
+   */
+  id?: string;
+  /**
+   * Classes to add to the input.
+   */
+  className?: string;
+  /**
+   * Name attribute for the input.
+   */
+  name: string;
+  /**
+   * Type attribute for the input.
+   */
+  type?: TextInputType;
+  /**
+   * Value of the input.
+   */
+  value: string;
+  /**
+   * Options for the {@link Label} component.
+   */
+  label: LabelProps;
+  /**
+   * Options for the {@link Hint} component (e.g. text).
+   */
+  hint?: HintProps;
+  /**
+   * Options for the {@link ErrorMessage} component (e.g. text).
+   */
+  errorMessage?: ErrorMessageProps;
+  /**
+   * Options for the {@link FormGroup} wrapper.
+   */
+  formGroup?: FormGroupPropsWithoutChildren;
+  /**
+   * Function to perform when the onChange event is fired.
+   */
+  onChange?(value: string): void;
+  /**
+   * Attribute to
+   * {@link https://www.w3.org/WAI/WCAG21/Understanding/identify-input-purpose.html|identify input purpose}
+   * for instance "postal-code" or "username". See
+   * {@link https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#autofill|autofill})
+   * for full list of attributes that can be used.
+   */
+  autocomplete?: string;
+  /**
+   * Set to `true` if this is a required field.
+   */
+  required?: boolean;
+  /**
+   * If `true`, input will be disabled.
+   */
+  disabled?: boolean;
+  /**
+   * Attribute to {@link https://www.w3.org/TR/html51/sec-forms.html#the-pattern-attribute provide a regular expression pattern}, used to match allowed character combinations for the input value
+   */
+  pattern?: string;
+}
+
+export const TextInput: React.FunctionComponent<TextInputProps> = props => {
```

Comment:
Is this actually just `Input`?

Reply:
Haha I just changed it from `Input` to `TextInput` because I thought that was more relevant/appropriate. In theory Input could be a radio or a date or hidden or a bunch of other stuff not covered by this component.

Reply:
Hrmm. Yeah. I see what you mean. But it's not just for text either... I think I prefer `Input` to `TextInput` personally, as it's probably what I'd reach for for an input box, but is there a better name?

Reply:
Yeah I see what you mean. I've changed back to Input now.